### ISSUE TITLE
[Feature Request]: Support multi-output models

### ISSUE THREAD

**Issue by Utanapishtim31** [SciSharp/TensorFlow.NET#1199]
### Background and Feature Description

It is not possible to train a model with several outputs. This is a problem for Actor-Critic models for instance.
This is a serious limitation of Tensorflow.NET for the time being.

### API Definition and Usage

```cs
IModel.compile(IOptimizer, ILossFunc[], IMetricFunc[])
```

### Risks

This does not seem to be very complex to add this feature. Most of the code already exists to handle multi-output.
One should compute the loss for each output and sum-reduce the losses to get the final loss.
